### PR TITLE
Feature: Apollo uplink: connection timeouts

### PR DESCRIPTION
**Pull request by o0Ignition0o**
This PR adds a timeout to apollo uplink schema polling (30s by default, configurable) and reports (90s, not configurable).

### CODE REVIEW

**Review by Geal — APPROVED** [review]

**Comment by o0Ignition0o** [comment]
I m going to rename APOLLO_UPLINK_CLIENT_TIMEOUT to APOLLO_UPLINK_TIMEOUT

**Review by BrynCooke — APPROVED** [review]
Approved, but let's remove `client` from `apollo-uplink-client-timeout` to give `apollo-uplink-timeout`

File: apollo-router/src/plugins/telemetry/apollo_exporter.rs [hunk 1]
```diff
@@ -35,6 +35,10 @@ const DEFAULT_QUEUE_SIZE: usize = 65_536;
 // Do not set to 5 secs because it's also the default value for the BatchSpanProcesser of tracing.
 // It's less error prone to set a different value to let us compute traces and metrics
 pub(crate) const EXPORTER_TIMEOUT_DURATION: Duration = Duration::from_secs(6);
+// Set to 90 seconds to err on the side of caution. This + our backoffs and attemps
+// should be more than enough to make sure the reports are sent,
+// while preventing potential hangs.
+pub(crate) const HTTP_CLIENT_TIMEOUT_DURATION: Duration = Duration::from_secs(90);
```
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/apollo_exporter.rs:41]: In documentation you mentioned 30 secs

Comment (bnjjj) [reply apollo-router/src/plugins/telemetry/apollo_exporter.rs:41]: My bad nvm

**Review by bnjjj — APPROVED** [review]

**Pull request merged**